src/lib/serviceRoutinesV2/optionsPostOnly.cpp

```diff
@@ -45,8 +46,11 @@ std::string optionsPostOnly
   ParseData*                 parseDataP
 )
 {
-  ciP->httpHeader.push_back("Access-Control-Allow-Methods");
-  ciP->httpHeaderValue.push_back("POST, OPTIONS");
+  if ( (ciP->httpHeaders.origin != "") && ((strcmp(corsOrigin, "__ALL") == 0) || (strcmp(ciP->httpHeaders.origin.c_str(), corsOrigin) == 0)) )

```

Comment:
This check is done many times... Maybe is should be moved to a common function.

Reply:
Which file do you think should host the function? Is `rest.cpp` a good candidate?

Reply:
`rest.cpp` (along with extern declaration in `rest.h`) sounds good.

Reply:
Fixed in 3e6e50a